tests/serialization/test_dag_serialization.py

```diff
@@ -543,6 +543,66 @@ def test_dag_serialized_fields_with_schema(self):
         dag_params: set = set(dag_schema.keys()) - ignored_keys
         self.assertEqual(set(DAG.get_serialized_fields()), dag_params)
 
+    def test_no_new_fields_added_to_base_operator(self):
+        """
+        This test verifies that there are no new fields added to BaseOperator. And reminds that
+        tests should be added for it.
+        """
+        base_operator = BaseOperator(task_id="10")
+        fields = base_operator.__dict__
+        self.assertEqual({'_dag': None,
+                          '_downstream_task_ids': set(),
+                          '_inlets': [],
+                          '_log': base_operator.log,
+                          '_outlets': [],
+                          '_upstream_task_ids': set(),
+                          'depends_on_past': False,
+                          'do_xcom_push': True,
+                          'email': None,
+                          'email_on_failure': True,
+                          'email_on_retry': True,
+                          'end_date': None,
+                          'execution_timeout': None,
+                          'executor_config': {},
+                          'inlets': [],
+                          'max_retry_delay': None,
+                          'on_execute_callback': None,
+                          'on_failure_callback': None,
+                          'on_retry_callback': None,
+                          'on_success_callback': None,
+                          'outlets': [],
+                          'owner': 'airflow',
+                          'params': {},
+                          'pool': 'default_pool',
+                          'priority_weight': 1,
+                          'queue': 'default',
+                          'resources': None,
+                          'retries': 0,
+                          'retry_delay': timedelta(0, 300),
+                          'retry_exponential_backoff': False,
+                          'run_as_user': None,
+                          'sla': None,
+                          'start_date': None,
+                          'subdag': None,
+                          'task_concurrency': None,
+                          'task_id': '10',
+                          'trigger_rule': 'all_success',
+                          'wait_for_downstream': False,
+                          'weight_rule': 'downstream'}, fields,
+                         """
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+     ACTION NEEDED! PLEASE READ THIS CAREFULLY AND CORRECT TESTS CAREFULLY
+
+ Some fields were added to the BaseOperator! Please add them to the list above and make sure that
+ you add support for DAG serialization - you should add the field to
+ `airflow/serialization/schema.json` and add it in `serialized_simple_dag_ground_truth` above
```

Comment:
Having said that I'm now not sure this last part is true.

So since DAG serialization schema version 1.0 is now released with 1.10.7 we might have to start thinking about versioning of this schema and migrate/update. But the important bit is that a row that exists in the DB right now must continue to work and that is what the "ground truth" (not the best name) is meant to represent.

Reply:
I think then we need to decide what to do in this case and tell @lokeshlal what to do in this case: https://github.com/apache/airflow/pull/7162 (and update this description).

Reply:
So the goal of the ground truth test is to ensure that DAGs that are _currently_ in people's databases get correctly handled as the model changes. The important bit I think is that this data can't change sa it's fixed in a DB.

_But_ some changes such as adding an optional field to Operators/Tasks is allowed, as the JSON Schema allows unknown fields at the task level.

But by default BaseOperator.get_serialized_fields will include extra fields. So I guess the only check I would like here is that new fields from base operator get specified with a type in the JSON schema, and that it sohuld be optional (if it's not optional then we would have to bump the version field in our schema. But we haven't worked out how to handle versioning of schemas and blobs yet!) -- which is sadly harder to test for.

Reply:
Similar rules as for protobuf. I will modify the message. We cannot test a lot of that fully automatically yet but we can at least provide the right message.

Reply:
I think this should good enough for the time being

Reply:
Oh yes, and switch away from JSON Schema. Geez its 2020 :-P